datastore/longhorn.go

```diff
@@ -1879,6 +1879,10 @@ func (s *DataStore) ListPodsRO() ([]*corev1.Pod, error) {
 	return s.pLister.Pods(s.namespace).List(labels.Everything())
 }
 
+func (s *DataStore) GetPodRO(namespace, name string) (*corev1.Pod, error) {
```

Comment:
This is good enough for now, just leaving a note here for the future.

NIT: there is kubernetes.go that contains functions for working with the kubernetes resources. Which is a part of the implementation of the datastore. It would require some additional cleanup, to not rely on the s.namespace instead as you have done here to pass the namespace explicitly.

Reply:
NIT: I think it's fine to add one more commit in which this function can be moved to `datastore/kubernetes.go`. And it's unnecessary to open a separate PR for this movement.